Sources/Frostflake/FrostflakeDefinitions.swift

```diff
@@ -17,15 +17,79 @@ public extension FrostflakeIdentifier {
     }
 }
 
-public extension Frostflake {
+// Internal Layout
+@usableFromInline
+enum FrostflakeLayout {
+    /// Default number of bits allocated to generator part, default 11 bits, 2.048 unique concurrent generators in the system
+    @usableFromInline
+    static let generatorIdentifierBits = 11
+
     /// Default number of bits allocated to seconds, default 32 bits, gives us ~136 years
+    @usableFromInline
     static let secondsBits = 32
 
     /// Default number of bits allocated to sequence, default 21 bits, 2.097.152 id:s max per second
+    @usableFromInline
     static let sequenceNumberBits = 21
 
+    /// <generatorIdentifierBits: 11><secondsBits: 32><sequenceNumberBits: 21>
+
+    private static let sequenceNumberBitsLayout = 0 ..< sequenceNumberBits
+    private static let secondsBitsLayout = sequenceNumberBitsLayout.upperBound ..< sequenceNumberBitsLayout.upperBound + secondsBits
+    private static let generatorIdentifierBitsLayout = secondsBitsLayout.upperBound ..< secondsBitsLayout.upperBound + generatorIdentifierBits
+
+    /// The range of valid generator identifiers
+    @usableFromInline
+    static let validGeneratorIdentifierRange = 0 ..< (1 << generatorIdentifierBits)
+
+    /// The range of valid sequence numbers
+    @usableFromInline
+    static let allowedSequenceNumberRange = 0 ..< (1 << sequenceNumberBits)
+
+    /// Convenience default manual generator identifier for the command line utility will pick the highest available identifier
+    @usableFromInline
+    static let defaultManualGeneratorIdentifier = (1 << generatorIdentifierBits) - 1
+
+    @usableFromInline
+    static func composeIdentifier(generatorId: UInt16, seconds: UInt32, seqNum: UInt32 = 0) -> UInt64 {
```

Comment:
Let's discuss this PR when I'm back from vacation, but I just want to point out that not only does this restructuring of the internal data structure cause a major version break for the package, it is also a major semantic break of all generated identifiers and would invalidate any existing data.

But more importantly, the choice of having the second bits in the beginning of the frost flake is very intentional to get good temporal sorting regardless of the originating generator id, now that it's being moved to the middle of the buffer that breaks (and may have performance implications).